### ISSUE TITLE
Re-add jRuby build

### ISSUE THREAD

**Issue by thijsc** [karafka/rdkafka-ruby#222]
I've removed the failing jRuby build for the moment so we can make progress on fixing the last remaining segfault. Is somebody interested in picking this up? The original contributor is not active anymore on Github.

**Comment by thijsc** [comment]
@Adithya-copart I believe you're using jRuby?

**Comment by hoprocker** [comment]
I've been the lead dev for the last 6 years on a project that uses zendesk/ruby-kafka, and so I'm sad that project is no longer maintained (but thanks for a great library, zendesk!). We're using Ruby+Kafka in a cross-platform library that runs under both MRI and JRuby, so having a unified API has been great. I'm looking around at other libraries that offer the same sort of clean, comprehensive interface that ruby-kafka does, and it looks like rdkafka is picking up momentum to be the framework-free successor. 

I'd be interested in digging into Java integration with this library. I have to say that I'm kind of surprised that there has been _any_ JRuby support, being as it's based on librdkafka. Before finding this ticket, my thought on the most direct way to add a native Java implementation was an `ext` branch tying into the official Apache client in a way that mirrors the existing rdkafka API. Looks like work has been done on using this library with JRuby already though, so I'd be interested in learning more about that.

**Comment by mensfeld** [comment]
@hoprocker happy to give you some hints.

The only thing aside of "just" doing it from the standpoint of expanding this library is to ensure there are no race conditions and concurrency issues. While I worked on removing the last memory leaks and providing stable behaviors, I did notice that in a few places, we potentially rely on having the GVL (not directly), and if we are supposed to re-introduce JRuby IMHO this is the crucial thing. I know I tackled some by introducing granular mutexes but I cannot promise you that I fixed all those things.

That said I would be happy to coop with you if you are willing to work on that.

**Comment by adi-pen** [comment]
@hoprocker It's been a few years but I used to work on a JRuby application that relied on `rdkafka-ruby`. It would mean adding C dependencies to your application.

> Before finding this ticket, my thought on the most direct way to add a native Java implementation was an ext branch tying into the official Apache client in a way that mirrors the existing rdkafka API. 

This is my thought process at the time but adding the native Java implementation is time-consuming and not something that I could commit to at the time. https://github.com/appsignal/rdkafka-ruby/pull/108 is where I took the initial stab of supporting JRuby.
